Sources/Diagnostics/FileHandler.swift

```diff
@@ -15,7 +15,29 @@ import Foundation
 
 /// A wrapper that allows basic operations on a file, synchronized as an `actor`.
 @available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, *)
-actor FileHandler {
+protocol FileHandlerType: Sendable {
+
+    /// - Note: this loads the entire file in memory
+    /// For newer versions, consider using `readLines` instead.
+    func readFile() async throws -> Data
+
+    /// Returns an async sequence for every line in the file
+    @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+    func readLines() async throws -> AsyncLineSequence<FileHandle.AsyncBytes>
+
+    /// Adds a line at the end of the file
+    func append(line: String) async
+
+    /// Removes the contents of the file
+    func emptyFile() async throws
+
+    /// Deletes the first N lines from the file, without loading the entire file in memory.
+    func removeFirstLines(_ count: Int) async throws
+
+}
+
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, *)
+actor FileHandler: FileHandlerType {
```

Comment:
Can't comment on the code that's not modified in the PR, but it looks great. The only way I can think of for improving it would be using a circular buffer for removing lines instead of an aux file to avoid extra disk reads

Reply:
which maybe we can revisit one day if we start having a lot of events

Reply:
The way removing lines is implemented is sort of like a circular buffer though (it reads and writes in chunks to prevent the entire file in memory all at once).